dockstore-integration-testing/src/test/java/io/dockstore/webservice/WebhookIT.java

```diff
@@ -359,6 +361,68 @@ public void testGitHubReleaseNoWorkflowOnDockstoreNoUser() throws Exception {
+        Workflow workflow = client.getWorkflowByPath("github.com/" + workflowRepo + "/foobar", "", false);
+        assertEquals("Should be a WDL workflow", Workflow.DescriptorTypeEnum.WDL, workflow.getDescriptorType());
+        assertEquals("Should be type DOCKSTORE_YML", Workflow.ModeEnum.DOCKSTORE_YML, workflow.getMode());
+        assertEquals("Should have one version", 1, workflow.getWorkflowVersions().size());
+        assertFalse("Should be invalid (wrong language, bad version)", workflow.getWorkflowVersions().get(0).isValid());
+
+        workflowsApi.updateDescriptorType(workflow.getId(), DescriptorLanguage.CWL.toString());
+        io.dockstore.openapi.client.model.Workflow updatedWorkflowAfterModifyingDescriptorType = workflowsApi.getWorkflow(workflow.getId(), "");
+        Assert.assertEquals("The descriptor language should have been changed", io.dockstore.openapi.client.model.Workflow.DescriptorTypeEnum.CWL, updatedWorkflowAfterModifyingDescriptorType.getDescriptorType());
+        Assert.assertEquals("The old versions should have been removed", 0, updatedWorkflowAfterModifyingDescriptorType.getWorkflowVersions().size());
+
+        workflowsApi.updateDescriptorType(workflow.getId(), DescriptorLanguage.WDL.toString());
+        updatedWorkflowAfterModifyingDescriptorType = workflowsApi.getWorkflow(workflow.getId(), "");
+        Assert.assertEquals("The descriptor language should have been changed", io.dockstore.openapi.client.model.Workflow.DescriptorTypeEnum.WDL, updatedWorkflowAfterModifyingDescriptorType.getDescriptorType());
+        Assert.assertEquals("The old versions should have been removed", 0, updatedWorkflowAfterModifyingDescriptorType.getWorkflowVersions().size());
+
+        // Release 0.1 on GitHub - one new wdl workflow
+        workflows = client.handleGitHubRelease(workflowRepo, BasicIT.USER_2_USERNAME, "refs/tags/0.1", installationId);
+        assertEquals("Should only have one service", 1, workflows.size());
+
+        // Ensure that new workflow is created and is what is expected
+        workflow = client.getWorkflowByPath("github.com/" + workflowRepo + "/foobar", "", false);
+        assertEquals("Should be a WDL workflow", Workflow.DescriptorTypeEnum.WDL, workflow.getDescriptorType());
+        assertEquals("Should be type DOCKSTORE_YML", Workflow.ModeEnum.DOCKSTORE_YML, workflow.getMode());
+        assertEquals("Should have one version 0.1", 1, workflow.getWorkflowVersions().size());
+        assertTrue("Should be valid", workflow.getWorkflowVersions().get(0).isValid());
+        try {
+            workflowsApi
+                    .updateDescriptorType(workflow.getId(), DescriptorLanguage.CWL.toString());
+            fail("Should not be able to change the descriptor type of a workflow that has valid versions");
+        } catch (io.dockstore.openapi.client.ApiException e) {
+            Assert.assertEquals("Cannot change descriptor type of a valid workflow", e.getMessage());
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'Assert.assertEquals' due to existing static import 'org.junit.Assert.assertEquals'](https://app.codacy.com/gh/dockstore/dockstore/pullRequest?prid=6225903)